src/foam/u2/crunch/CapabilityStore.js

```diff
@@ -467,14 +469,22 @@ foam.CLASS({
             this.IN('featured', this.Capability.KEYWORDS)
           )).select())
         .then(async sink => {
-          if ( sink.array.length == 1 ) {
+          if ( sink.array.length !== 1 )
+            return;
+
             let cap = sink.array[0];
             let ucj = await this.junctions.find(ucj => ucj.targetId == cap.id);
             if ( ucj && ( ucj.status == this.CapabilityJunctionStatus.GRANTED
               || ucj.status == this.CapabilityJunctionStatus.PENDING ) ) return;
 
-            this.openWizard(cap, false);
-          }
+            let x = this.ctrl.__subContext__;
+            await this.grantAll(x, cap.id, this.subject);
```

Comment:
don't think `await this.grantAll(x, cap.id, this.subject);` is needed.

Reply:
Seems It does need it. Otherwise not granted.  E.g. If one left visible cap is "crunch.onboarding.connect-general-admission" then "crunch.onboarding.general-admission" required too.

Reply:
with https://github.com/kgrgreer/foam3/pull/1550 we can set the property that Ruby added to the capability and all will be good...